### DISCUSSION TITLE
Support for old image (v7.0 - v.10)

### DISCUSSION THREAD

**Discussion by andhit-r** [Tecnativa/doodba#447] [Doodba - The image] [answered]
Hi,

I notice that from commit 20dfe8a42487ce00f8220d56e06263c296bfd6db doodba remote old image (v7.0 - v10.0) due test failing. Is there any plans to reintroduce old image? I would like to help if Tecnativa have a plan to reintroduce old image.

**Comment by pedrobaeza** [comment, marked as answer]
The images are there. What is not there is the CI.

**Reply by andhit-r** [reply]
Yes the image still there. What i meant is the Dockerfile, sorry.

**Reply by pedrobaeza** [reply]
But for what do you need to update such Dockerfile?

**Reply by andhit-r** [reply]
Let say something break like https://github.com/nodesource/distributions/issues/1266#issuecomment-932877073. Adjusting Dockerfile needed to workaround this issue temporarily. Otherwise docker-compose build will raise error. Since that issue start my dev and prod env could not build. I create https://hub.docker.com/repository/docker/osind/doodba temporarily. It would be nice if i can just push it to Tecnativa.

I could not apply any workaround with child project since the error is on ONBUILD instruction. So adjusting Dockerfile (rebuild image) is a must. But if you can suggest another workaround i'm all ears.

**Reply by pedrobaeza** [reply]
Each of these issues, which should be extremely rare, can be fixed in a exceptional way, but a constant rebuild taking care of all dependencies, OS images availability, and so, can't be kept continuously. Not sure if my colleagues @Yajo and @joao-p-marques have already a solution for the certificate problem.

**Reply by yajo** [reply]
@joao-p-marques is working on that for legacy images, but those fixes land as conditionally into doodba-copier-template, as it happened some time ago in https://github.com/Tecnativa/doodba-copier-template/pull/252.

**Reply by joao-p-marques** [reply]
Exactly. Doodba has some other mechanisms that allow us to patch older images without having to unfreeze them. The fix for this particular issue is already on the way with https://github.com/Tecnativa/doodba-copier-template/pull/279 and future ones should be similar.

**Reply by andhit-r** [reply]
Wow .. i did not realize doodba has this mechanism. Nice. It seems that i still need to explore doodba capabilities. 

I will wait for https://github.com/Tecnativa/doodba-copier-template/pull/279. Many thanks @joao-p-marques @pedrobaeza @Yajo

**Reply by joao-p-marques** [reply]
No problem! You can find more about it in https://github.com/Tecnativa/doodba#optodoocustombuildd

**Reply by andhit-r** [reply]
Many thanks @joao-p-marques.
